src/league.js

```diff
@@ -71,205 +84,122 @@ league_attributes = {
     _lastUpdated: moment.utc(),
 
     //--------------------------------------------------------------------------
-    // Whether this is the first time through updating or not.
-    //--------------------------------------------------------------------------
-    _firstRosterUpdate: true,
-
-    //--------------------------------------------------------------------------
     // Canonicalize the username
     //--------------------------------------------------------------------------
     canonicalUsername: function(username) {
         username = username.split(" ")[0];
         return username.replace("*", "");
     },
+    
+    //--------------------------------------------------------------------------
+    // Lookup a player by playerName
+    //--------------------------------------------------------------------------
+    getPlayer: function(username) {
+        return this._playerLookup[_.toLower(username)];
+    },
 
     //--------------------------------------------------------------------------
     // Refreshes everything
     //--------------------------------------------------------------------------
     'refresh': function() {
         var self = this;
-        self.refreshRosters(function(err, rosters) {
-            if (err) {
-                winston.error("Unable to refresh rosters: " + err);
-                throw new Error(err);
-            } else {
-                winston.info("Found " + rosters.length + " teams for " + self.options.name);
-            }
-            self._lastUpdated = moment.utc();
-        });
-        self.refreshCurrentRoundSchedules(function(err, pairings) {
-            if (err) {
-                winston.error("Unable to refresh schedule: " + err);
-                throw new Error(err);
-            } else {
+        return Q.all([
+            self.refreshRosters().then(function() {
```

Comment:
stylistically, this is much cleaner. good call on use of Q.all and promises here.